docs/plugin.md

```diff
@@ -0,0 +1,97 @@
+To add custom rules first extend the [Rule](https://github.com/Skyscanner/cfripper/blob/master/cfripper/model/rule.py)
+class. Then implement the `invoke` method by adding your logic.
+
+{{ inline_source('cfripper.rules.base_rules.Rule.invoke') }}
+
+CFRipper uses [pycfmodel](https://github.com/Skyscanner/pycfmodel) to create a Python model of the CloudFormation script.
+This model is passed to the `invoke` function as the `cfmodel` parameter. You can use the model's iterate through the
+resources and other objects of the model and use the helper functions to perform various checks. Look at the
+[current rules](/rules) for examples.
+
+## Making your rules available to CFRipper
+
+CFRipper in order to load dynamic rules uses [pluggy](https://github.com/pytest-dev/pluggy). You have to define a 
+so-called entry point for your distribution so that cfripper finds your plugin module. Entry points are a feature 
+provided by setuptools. CFRipper CLI looks up the `cfripper` entrypoint to discover its plugins and load them.
```

Comment:
```suggestion
CFRipper uses [pluggy](https://github.com/pytest-dev/pluggy) in order to load dynamic rules. You have to define an 
entrypoint for your distribution so that CFRipper finds your plugin module. Entrypoints are a feature 
provided by `setuptools`. The CFRipper CLI looks up the `cfripper` entrypoint to discover its plugins and load them.
```